drivers/syslog/syslog_filechannel.c

```diff
@@ -85,28 +89,44 @@ static void log_rotate(FAR const char *log_file)
       return;
     }
 
-  /* Construct the backup file name. */
+  /* Rotated file names. */
 
-  backup_file = malloc(strlen(log_file) + 3);
-  if (backup_file == NULL)
+  name_size = strlen(log_file) + 8;
+  rotate_to = malloc(name_size);
+  rotate_from = malloc(name_size);
+  if ((rotate_to == NULL) || (rotate_from == NULL))
     {
-      return;
+      goto end;
     }
 
-  sprintf(backup_file, "%s.0", log_file);
+  /* Rotate the logs. */
+
+  for (i = (CONFIG_SYSLOG_FILE_ROTATIONS - 1); i > 0; i--)
+    {
+      snprintf(rotate_to, name_size, "%s.%d", log_file, i);
+      snprintf(rotate_from, name_size, "%s.%d", log_file, i - 1);
+
+      if (access(rotate_to, F_OK) == 0)
+        {
+          remove(rotate_to);
+        }
+
+      rename(rotate_from, rotate_to);
+    }
 
-  /* Delete any old backup files. */
+  snprintf(rotate_to, name_size, "%s.0", log_file);
 
-  if (access(backup_file, F_OK) == 0)
+  if (access(rotate_to, F_OK) == 0)
     {
-      remove(backup_file);
+      remove(rotate_to);
     }
 
-  /* Rotate the log. */
+  rename(log_file, rotate_to);
```

Comment:
line 116-124 can be simplify to:
rename(log_file, rotate_from);

Reply:
I am not sure if I understand this correctly.  
You mean to take advantage of `rotate_from` as it is already set in the loop?

`CONFIG_SYSLOG_FILE_ROTATIONS` can be set to 1, so in this case rotate_from will not be set (as the loop will never execute).